src/vueRecaptcha.vue

```diff
@@ -62,19 +68,44 @@
 
     onMounted(() => {
         if (window.grecaptcha == null) {
-            new Promise<void>((resolve) => {
+            new Promise<void>((resolve, reject) => {
+                let loadingCountdown: ReturnType<typeof setTimeout>;
+                let responded = false;
+
                 window.recaptchaReady = function () {
+                    if (responded) return;
+                    responded = true;
+                    clearTimeout(loadingCountdown);
                     resolve();
                 };
 
-                const doc = window.document;
                 const scriptId = "recaptcha-script";
+                const loadingFailed = (reason: string) => {
+                    return () => {
+                        if (responded) return;
+                        responded = true;
+
+                        // remove failed script so we can retry again later
+                        document.getElementById(scriptId)?.remove();
+
+                        reject(reason);
+                    }
+                };
+
+                // wait for script to load and error out instead of blocking forever
+                if (props.loadingTimeout > 0) loadingCountdown = setTimeout(loadingFailed("timeout"), props.loadingTimeout);
+
+                const doc = window.document;
                 const scriptTag = doc.createElement("script");
                 scriptTag.id = scriptId;
-                scriptTag.setAttribute("src", `https://www.google.com/recaptcha/api.js?onload=recaptchaReady&render=explicit&hl=${props.hl}`);
+                scriptTag.onerror = loadingFailed("error");
+                scriptTag.onabort = loadingFailed("aborted");
+                scriptTag.setAttribute("src", `https://www.google.com/recaptcha/api.js?onload=recaptchaReady&render=explicit&hl=${props.hl}&_=${+new Date()}`);
```

Comment:
`+new Date()` what is + here?

Reply:
This just produces a unix timestamp in milliseconds. The idea is to create a unique URL so the browser will actually attempt to load it again if we retry after a failure.